### ISSUE TITLE
Is it normal to slightly exceed the boundary value？

### ISSUE THREAD

**Issue by willi7758521** [osqp/osqp#389]
Although with the non-negtive constrain,the solution of osqp is still have some negtive values, the range of them is about -10^(-2) to 0;
Is this phenomeno normal? Thanks

**Comment by goulart-paul** [comment]
Yes.   It will only satisfy constraints to within the termination criteria you set for the solver.  This means that when the solver stops you may well have small constraint violations still.

**Comment by tomwenseleers** [comment]
Would it not make sense to clip the returned coefficient estimates within the requested constraints?

**Comment by wadasdw88s** [comment]
same question!
how to aovid the situation?
